### PR TITLE
Add dense variant to logo section

### LINKED ISSUE #4642: Small version of the logo section
## Pattern to amend

Sometimes the number of logos or their relative importance doesn't justify using 2 columns per logo: 

For those cases, a version of the logo section that scales logos to a single column could be very useful. I propose adding ti with a `--small` modifier.

## Visual

image

## Context

When there are too many logos, or when they need to be toned down

### PR DESCRIPTION
**Pull request by abhigyanghosh30**
## Done

Added denser version of logo section where the logos span only a single column
Fixes #4642 

## QA

- Open [demo](insert-demo-url)
- [Add QA steps]
- Review updated documentation:
  - [List any updated documentation for review]

### Check if PR is ready for release

If this PR contains Vanilla SCSS code changes, it should contain the following changes to make sure it's ready for the release:

- [x] PR should have one of the following labels to automatically categorise it in release notes:
  - `Feature 🎁`, `Breaking Change 💣`, `Bug 🐛`, `Documentation 📝`, `Maintenance 🔨`.
- [x] Vanilla version in `package.json` should be updated relative to the [most recent release](https://github.com/canonical/vanilla-framework/releases/latest), following semver convention:
  - if CSS class names are not changed it can be bugfix relesase (x.x.**X**)
  - if CSS class names are changed/added/removed it should be minor version (x.**X**.0)
  - see the [wiki for more details](https://github.com/canonical/vanilla-framework/wiki/Release-process#pre-release-tasks)
- [x] Any changes to component class names (new patterns, variants, removed or added features) should be listed on the [what's new page](https://github.com/canonical/vanilla-framework/blob/main/templates/docs/whats-new.md).
- [x] Documentation side navigation should be updated with the relevant labels.

## Screenshots

[if relevant, include a screenshot or screen capture]

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -79,4 +79,4 @@
     "stylelint-prettier": "2.0.0",
     "svgo": "2.8.0"
   }
-}
+}
```
Comment (bartaz) [inline package.json:82]: @abhigyanghosh30 I think linter complains about this one, does your editor remove new lines on save or something?

**Review by bartaz — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (abhigyanghosh30) [reply package.json]: Oh prettier was not configured properly on my setup. It did not work on .json files

**Review by bartaz — APPROVED** [review]
LGTM, thanks!

File: templates/docs/examples/patterns/logo-section/logo-section-dense.html [hunk 2]
```diff
@@ -0,0 +1,28 @@
+{% extends "_layouts/examples.html" %}
+{% block title %}Logo section Dense{% endblock %}
```
Comment (bartaz) [inline templates/docs/examples/patterns/logo-section/logo-section-dense.html:2]: Small fix for our title convention

```suggestion
{% block title %}Logo section / Dense{% endblock %}
```

**Pull request merged**